From 0c0b1efd6e194562cbc6c37530dee6bbdbd6a604 Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Tue, 28 Apr 2015 23:53:57 +0200
Subject: [PATCH] Fix configure check for libsystemd

Add support for building against libsystemd, which has superseded
libsystemd-login [1]. We keep a fallback to the old libsystemd-login
library though and also keep the name for the configure switch.

Also, error out, if libsystemd support is requested but the library
could not be found.

[1] http://lists.freedesktop.org/archives/systemd-devel/2014-February/017146.html

Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=779742
---
 configure.ac | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

Index: gnome-disk-utility/configure.ac
===================================================================
--- gnome-disk-utility.orig/configure.ac	2015-04-29 00:03:45.741090442 +0200
+++ gnome-disk-utility/configure.ac	2015-04-29 00:03:45.737090399 +0200
@@ -130,20 +130,25 @@
 dnl *** Check for libsystemd-login ***
 dnl **********************************
 
-AC_ARG_ENABLE(libsystemd_login, AS_HELP_STRING([--disable-libsystemd-login],[build without liblibsystemd-login]))
+AC_ARG_ENABLE(libsystemd_login, AS_HELP_STRING([--disable-libsystemd-login],[build without libsystemd-login]))
 msg_libsystemd_login=no
-LIBSYSTEMD_LOGIN_LIBS=
-LIBSYSTEMD_LOGIN_CFLAGS=
-LIBSYSTEMD_LOGIN_REQUIRED=44
 
-if test "x$enable_libsystemd_login" != "xno"; then
-  PKG_CHECK_EXISTS([libsystemd-login >= $LIBSYSTEMD_LOGIN_REQUIRED], msg_libsystemd_login=yes)
+if test "$enable_libsystemd_login" != "no"; then
+  PKG_CHECK_MODULES([LIBSYSTEMD_LOGIN],[libsystemd >= 209], [msg_libsystemd_login=yes],
+                    [PKG_CHECK_MODULES(LIBSYSTEMD_LOGIN, [libsystemd-login >= 44],
+                    [msg_libsystemd_login=yes],
+                    [msg_libsystemd_login=no])])
 
-  if test "x$msg_libsystemd_login" = "xyes"; then
-    PKG_CHECK_MODULES([LIBSYSTEMD_LOGIN],[libsystemd-login >= $LIBSYSTEMD_LOGIN_REQUIRED])
+  if test "$msg_libsystemd_login" = "yes"; then
     AC_DEFINE(HAVE_LIBSYSTEMD_LOGIN, 1, [Define to 1 if liblibsystemd_login is available])
+  else
+    if test "$enable_libsystemd_login" = "yes"; then
+      AC_MSG_ERROR([libsystemd support requested but libsystemd or libsystemd-login library not found])
+    fi
   fi
 fi
+AC_SUBST(LIBSYSTEMD_LOGIN_LIBS)
+AC_SUBST(LIBSYSTEMD_LOGIN_CFLAGS)
 
 AM_CONDITIONAL(USE_LIBSYSTEMD_LOGIN, [test "$msg_libsystemd_login" = "yes"])
 
@@ -225,7 +230,7 @@
         sysconfdir:                 ${sysconfdir}
         localstatedir:              ${localstatedir}
 
-        Use libsystem-login:        ${msg_libsystemd_login}
+        Use libsystemd-login:       ${msg_libsystemd_login}
         Build g-s-d plug-in:        ${msg_gsd_plugin}
 
         compiler:                   ${CC}
